app/controllers/prison/visits_controller.rb

```diff
@@ -4,15 +4,11 @@ class Prison::VisitsController < ApplicationController
   before_action :require_login_during_trial,
     only: %w[show nomis_cancelled process_visit update]
   before_action :cancellation_reason_set, only: :cancel
+  before_action :visit_is_processable, only: [:process_visit, :update]
```

Comment:
This could be done inside the ```process_visit``` and ```update``` actions as it requires the ```@visit``` to be loaded and we already do that in each action.